kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java

```diff
@@ -56,15 +57,27 @@ public KytheDocTreeScanner(KytheTreeScanner treeScanner, Context context) {
     this.trees = JavacTrees.instance(context);
   }
 
-  public boolean visitDocComment(TreePath treePath, VName node, EntrySet absNode) {
+  @AutoValue
+  abstract static class DocCommentVisitResult {
+    abstract boolean documented();
+
+    abstract Optional<String> deprecation();
+
+    static DocCommentVisitResult create(boolean documented, Optional<String> deprecation) {
```

Comment:
Let's optimize a bit. Have a static constant UNDOCUMENT, and one creation method that takes the Optional<String>.

Reply:
Done